File: 5-1.py
import cv2 as cv
import sys




img_color = cv.imread('copy.png', cv.IMREAD_COLOR)

if img_color is None:
    print('can not read image file')
    sys.exit(1)

img_gray = cv.cvtColor(img_color, cv.COLOR_BGR2GRAY)

img_binary = cv.adaptiveThreshold(img_gray, 255, cv.ADAPTIVE_THRESH_MEAN_C,cv.THRESH_BINARY_INV, 5,4)

cv.imshow('GrayScale', img_gray)
cv.imshow('Binary', img_binary)
cv.waitKey(0)

cv.destroyAllWindows()

File: 5-1.cpp
#include "iostream"
#include "opencv2/opencv.hpp"

using namespace cv;
using namespace std;

int main(){

    Mat img_color = imread("copy.png", IMREAD_COLOR);
    if(img_color.empty()){
        cout << "can not read image" << endl;
        return -1;
    }
    
    Mat img_gray;
    cvtColor(img_color, img_gray, COLOR_BGR2GRAY);

    Mat img_binary;
    adaptiveThreshold(img_gray, img_binary, 255, ADAPTIVE_THRESH_MEAN_C, THRESH_BINARY, 5,4);

    imshow("GrayScale", img_gray);
    imshow("Binary", img_binary);
    waitKey(0);

    destroyAllWindows();
    return 0;
}